modules/repository-url/src/main/java/org/elasticsearch/plugin/repository/url/URLRepositoryPlugin.java

```diff
@@ -40,6 +58,41 @@
                                                            ClusterService clusterService, BigArrays bigArrays,
                                                            RecoverySettings recoverySettings) {
         return Collections.singletonMap(URLRepository.TYPE,
-            metadata -> new URLRepository(metadata, env, namedXContentRegistry, clusterService, bigArrays, recoverySettings));
+            metadata ->  {
+            assert httpClient.get() != null : "Expected to get a configured http client";
+            return new URLRepository(metadata, env, namedXContentRegistry, clusterService, bigArrays, recoverySettings, httpClient.get());
+        });
+    }
+
+    @Override
+    public Collection<Object> createComponents(Client client,
+                                               ClusterService clusterService,
+                                               ThreadPool threadPool,
+                                               ResourceWatcherService resourceWatcherService,
+                                               ScriptService scriptService,
+                                               NamedXContentRegistry xContentRegistry,
+                                               Environment environment,
+                                               NodeEnvironment nodeEnvironment,
+                                               NamedWriteableRegistry namedWriteableRegistry,
+                                               IndexNameExpressionResolver indexNameExpressionResolver,
+                                               Supplier<RepositoriesService> repositoriesServiceSupplier) {
+
+        PoolingHttpClientConnectionManager connManager = new PoolingHttpClientConnectionManager();
+        final CloseableHttpClient apacheHttpClient = HttpClients.custom()
+            .disableAutomaticRetries()
+            .setConnectionManager(connManager)
+            .build();
+        final URLHttpClient apacheURLHttpClient = new URLHttpClient(apacheHttpClient, connManager);
+        httpClient.compareAndSet(null, apacheURLHttpClient);
+        return List.of(apacheURLHttpClient);
+    }
+
+    @Override
+    public void close() throws IOException {
+        super.close();
+        URLHttpClient urlHttpClient = httpClient.getAndSet(null);
+        if (urlHttpClient != null) {
+            IOUtils.closeWhileHandlingException(urlHttpClient);
```

Comment:
closeWhileHandlingException can handle null values, no need for null check

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/69521/commits/a78eb96741817e1d4bcc977a222cf7d9f71ca047